src/Middleware/Localization/src/RequestLocalizationMiddleware.cs

```diff
@@ -26,22 +26,24 @@ public class RequestLocalizationMiddleware
 
         private readonly RequestDelegate _next;
         private readonly RequestLocalizationOptions _options;
-        private ILogger _logger;
+        private readonly ILogger<RequestLocalizationMiddleware> _logger;
 
         /// <summary>
         /// Creates a new <see cref="RequestLocalizationMiddleware"/>.
         /// </summary>
         /// <param name="next">The <see cref="RequestDelegate"/> representing the next middleware in the pipeline.</param>
         /// <param name="options">The <see cref="RequestLocalizationOptions"/> representing the options for the
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> used for logging.</param>
         /// <see cref="RequestLocalizationMiddleware"/>.</param>
-        public RequestLocalizationMiddleware(RequestDelegate next, IOptions<RequestLocalizationOptions> options)
+        public RequestLocalizationMiddleware(RequestDelegate next, IOptions<RequestLocalizationOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Why did you take the change in https://github.com/aspnet/AspNetCore/pull/10946/commits/16739e0cfdbff8c4b50ea3398da4b7d9f585c561?

So to not inject the `ILogger<RequestLocalizationMiddleware>`.

Reply:
I saw that other middlewares inject `ILoggerFactory` so I wanted to keep the same pattern

Reply:
The breaking change guidelines are to add a new constructor and obsolete the old one. For weird DI reasons the new constructor should be placed above the old one.

Reply:
So how should we create logger object for the old constructor. Maybe I should put back the EnsureLogger!!?

Reply:
So how should we create logger object for the old constructor. Maybe I should put back the EnsureLogger!!?

Reply:
> For weird DI reasons the new constructor should be placed above the old one.

Does this actually matter to DI? the order used by reflection is not a stable guarantee 😢

Reply:
It does for the ActivatorUtilities library we use to construct middleware. Middleware aren't in the DI container so we build them manually and pull their dependencies from DI.

Reply:
There's an attribute that you can use to specify the constructor to be used - `ActivatorUtilitiesConstructorAttribute`

Reply:
@Kahbazi here's what I'd do:

1) Add a new ctor which accepts a `ILoggerFactory` \ `ILogger<T>` instance. I personally prefer the latter, but there isn't a clear guidance for one over the other.
2) Add `[Obsolete]` attribute to the current attribute. Add `ActivatorUtilitiesConstructorAttribute` to the new ctor
3) Change the current constructor to call the other ctor. Have it pass a `NullLogger` instance. This should avoid the need to have `EnsureLogger`

Reply:
@Kahbazi here's an example, we use a NullLogger for the other constructor:
https://github.com/aspnet/AspNetCore/blob/62351067ff4c1401556725b401478e648b66acdc/src/Security/CookiePolicy/src/CookiePolicyMiddleware.cs#L20-L32

Reply:
So the log will be disabled if someone directly uses the old constructor, is that ok?

Reply:
Yes, it's quite unlikely anyone will use the old one, but we leave it for binary compat.